src/RocketChatChatProvider/src/RocketChatChatProvider/Controllers/AreaController.cs

```diff
@@ -1,13 +1,31 @@
 ﻿using ChatProvider.Models;
 using Microsoft.AspNetCore.Mvc;
+using RocketChatChatProvider.Client;
 
 namespace RocketChatChatProvider.Controllers
 {
     public class AreaController : ChatProvider.Controllers.AreaApiController
     {
-        public override Task<IActionResult> CreateArea([FromBody] CreateAreaRequest? createAreaRequest)
+        private readonly ILogger<AreaController> _logger;
+        private IRocketChatClient _client;
+
+        public AreaController(ILogger<AreaController> logger, IRocketChatClient client)
+        {
+            _logger = logger;
+            _client = client;
+        }
+
+        public override async Task<IActionResult> CreateArea([FromBody] CreateAreaRequest? createAreaRequest)
         {
-            throw new NotImplementedException();
+            if (createAreaRequest == null || string.IsNullOrEmpty(createAreaRequest.Name))
+            {
+                return BadRequest(createAreaRequest);
+            }
+
+            var name = NameUtil.SanitizeName(createAreaRequest.Name);
+            var area = await _client.CreateArea(name);
```

Comment:
Jag tror att _client kan kasta exception som dborde fångas och loggas och returnera något generiskt Problem objekt istället.

Reply:
yes, lade även in BadRequest om createarerequest är null eller om det har tomt namn